### PR TITLE
prep os.c for conversion to D

### PR DESCRIPTION
**Pull request by thewilsonator**
I think this is missing a couple of narrowing casts, but w/e.

### CODE REVIEW

File: src/dmd/backend/os.c [hunk 1]
```diff
@@ -161,7 +169,7 @@ void *globalrealloc(void *oldp,size_t newsize)
  * Functions to manage allocating a single virtual address space.
  */
 
-void *vmem_reserve(void *ptr,unsigned long size)
+void *vmem_reserve(void *ptr,ulong size)
```
Comment (WalterBright) [inline src/dmd/backend/os.c]: This is not correct. A C `unsigned long` is a `uint` in D, not a `ulong`.

File: src/dmd/backend/os.c [hunk 2]
```diff
@@ -183,17 +191,17 @@ void *vmem_reserve(void *ptr,unsigned long size)
  *      !=0     success
  */
 
-int vmem_commit(void *ptr, unsigned long size)
+int vmem_commit(void *ptr, ulong size)
```
Comment (WalterBright) [inline src/dmd/backend/os.c]: `uint`

File: src/dmd/backend/os.c [hunk 3]
```diff
@@ -183,17 +191,17 @@ void *vmem_reserve(void *ptr,unsigned long size)
...
     dbg_printf("vmem_commit(ptr = %p,size = x%lx)\n",ptr,size);
-    i = (int) VirtualAlloc(ptr,size,MEM_COMMIT,PAGE_READWRITE);
+    i = cast(int) VirtualAlloc(ptr,size,MEM_COMMIT,PAGE_READWRITE);
     if (i == 0)
         dbg_printf("failed to commit\n");
     return i;
 }
 
-void vmem_decommit(void *ptr,unsigned long size)
+void vmem_decommit(void *ptr,ulong size)
```
Comment (WalterBright) [inline src/dmd/backend/os.c]: `uint`

File: src/dmd/backend/os.c [hunk 4]
```diff
@@ -202,7 +210,7 @@ void vmem_decommit(void *ptr,unsigned long size)
     }
 }
 
-void vmem_release(void *ptr,unsigned long size)
+void vmem_release(void *ptr, ulong size)
```
Comment (WalterBright) [inline src/dmd/backend/os.c]: `uint`

File: src/dmd/backend/os.c [hunk 5]
```diff
@@ -225,13 +233,13 @@ void vmem_release(void *ptr,unsigned long size)
...
-static void *preserve;
-static size_t preserve_size;
+private __gshared HANDLE hFile = INVALID_HANDLE_VALUE;
+private __gshared HANDLE hFileMap = null;
+private __gshared void *pview;
+private __gshared void *preserve;
+private __gshared size_t preserve_size;
 
-void *vmem_mapfile(const char *filename,void *ptr,unsigned long size,int flag)
+void *vmem_mapfile(const char *filename,void *ptr, ulong size,int flag)
```
Comment (WalterBright) [inline src/dmd/backend/os.c]: `uint`

File: src/dmd/backend/os.c [hunk 6]
```diff
@@ -304,31 +312,31 @@ void *vmem_mapfile(const char *filename,void *ptr,unsigned long size,int flag)
...
-    return NULL;                        // failure
+    return null;                        // failure
 }
 
 /*****************************
  * Set size of mapped file.
  */
 
-void vmem_setfilesize(unsigned long size)
+void vmem_setfilesize(ulong size)
```
Comment (WalterBright) [inline src/dmd/backend/os.c]: `uint`

File: src/dmd/backend/os.c [hunk 7]
```diff
@@ -414,7 +422,7 @@ void *vmem_baseaddr()
  * *psize downwards.
  */
 
-void vmem_reservesize(unsigned long *psize)
+void vmem_reservesize(ulong *psize)
```
Comment (WalterBright) [inline src/dmd/backend/os.c]: `uint`

File: src/dmd/backend/os.c [hunk 8]
```diff
@@ -443,7 +451,7 @@ void vmem_reservesize(unsigned long *psize)
             size = (ms.dwAvailPageFile < ms.dwAvailVirtual)
                 ? ms.dwAvailPageFile
                 : ms.dwAvailVirtual;
-            size = (unsigned long long)size * 8 / 10;
+            size = cast(ulong)size * 8 / 10;
             size &= ~0xFFFFL;
```
Comment (WalterBright) [inline src/dmd/backend/os.c]: `L` suffix in D means `long`, not `int`

**Comment by WalterBright** [comment]
This has problems (see my review comments). Please fix.

**Comment by thewilsonator** [comment]
Will do shortly.

**Comment by thewilsonator** [comment]
#8949

File: src/dmd/backend/os.c (hunk 1 above)
Comment (jacob-carlborg) [reply src/dmd/backend/os.c]: `unsigned long` should be `core.stdc.config.c_ulong`. `long` in C is sometimes 32bit and sometimes 64bit, deepening on which platform. @thewilsonator ^

Comment (thewilsonator) [reply src/dmd/backend/os.c]: this function and I think the vast majority of this file is completely unused.

Comment (thewilsonator) [reply src/dmd/backend/os.c]: Hence why despite being wrong went through green. I plan to do a dead strip of DMD at some point.

**Pull request merged**